fate/src/main/java/org/apache/accumulo/fate/AdminUtil.java

```diff
@@ -235,10 +266,40 @@ public FateStatus getStatus(ReadOnlyTStore<T> zs, IZooReader zk, String lockPath
           pos++;
         }
 
-      } catch (Exception e) {
-        log.error("Failed to read locks for " + id + " continuing.", e);
+      } catch (KeeperException ex) {
+        /*
+         * could be transient zk error. Log, but try to process rest of list rather than throwing
+         * exception here
+         */
+        log.error("Failed to read locks for " + id + " continuing.", ex);
+
+      } catch (InterruptedException ex) {
+        Thread.currentThread().interrupt();
+        throw ex;
       }
     }
+  }
+
+  /**
+   * Returns fate status, possibly filtered
+   *
+   * @param zs
+   *          read-only access to a populated transaction store.
+   * @param filterTxid
+   *          Optional. List of transactions to filter results - if null, all transactions are
+   *          returned
+   * @param filterStatus
+   *          Optional. List of status types to filter results - if null, all transactions are
+   *          returned.
+   * @param heldLocks
+   *          populated list of locks held by transaction - or an empty map if none.
+   * @param waitingLocks
+   *          populated list of locks held by transaction - or an empty map if none.
+   * @return current fate and lock status
+   */
+  public FateStatus getTransactionStatus(ReadOnlyTStore<T> zs, Set<Long> filterTxid,
```

Comment:
Seems like it may make sense to make this private and have something like the following two public methods that call it.

```java
 public FateStatus getStatus(ReadOnlyTStore<T> zs, IZooReader zk, String lockPath, Set<Long> filterTxid, EnumSet<TStatus> filterStatus)
  public List<TransactionStatus> getTransactionStatus(ReadOnlyTStore<T> zs, IZooReader zk, Set<Long> filterTxid, EnumSet<TStatus> filterStatus)
```